zerver/lib/request.py

```diff
@@ -366,3 +367,16 @@ def _wrapped_view_func(request: HttpRequest, *args: object, **kwargs: object) ->
         return view_func(request, *args, **kwargs)
 
     return cast(ViewFuncT, _wrapped_view_func)  # https://github.com/python/mypy/issues/1927
+
+
+local = threading.local()
+
+def get_current_request() -> Optional[HttpRequest]:
+    return getattr(local, 'request', None)
+
+def set_request(req: HttpRequest) -> None:
+    setattr(local, 'request', req)
+
+def unset_request() -> None:
+    if hasattr(local, 'request'):
+        delattr(local, 'request')
```

Comment:
This sort of global is OK to exist since I see the utility for Sentry, but I think we should add comments and/or adjust naming to indicate that these should only be used for Sentry error-handling code paths and that our general code pattern is to pass `request` objects to functions.

Reply:
It's used even more, actually, for the non-Sentry codepath.  I'll put some comments in that this should be used for logging codepaths only.